src/red_black_tree.jl

```diff
@@ -0,0 +1,380 @@
+end
+
+"""
+    delete_fix(tree::RBTree, node::Union{RBTreeNode, Nothing})
+
+This method is called when a black node is deleted because it violates the black depth property of the RBTree. 
+"""
+function delete_fix(tree::RBTree, node::Union{RBTreeNode, Nothing})
+    while node != tree.root && !node.color 
+         if node == node.parent.leftChild
+            sibling = node.parent.rightChild
+            if sibling.color
+                sibling.color = false
+                node.parent.color = true
+                left_rotate!(tree, node.parent)
+                sibling = node.parent.rightChild 
+            end
+
+            if !sibling.rightChild.color && !sibling.leftChild.color
+                sibling.color = true
+                node = node.parent
+            else
+                if !sibling.rightChild.color
+                    sibling.leftChild.color = false
+                    sibling.color = true
+                    right_rotate!(tree, sibling)
+                    sibling = node.parent.rightChild
+                end
+
+                sibling.color = node.parent.color
+                node.parent.color = false
+                sibling.rightChild.color = false
+                left_rotate!(tree, node.parent)
+                node = tree.root
+            end
+        else
+            sibling = node.parent.leftChild
+            if sibling.color
+                sibling.color = false
+                node.parent.color = true
+                right_rotate!(tree, node.parent)
+                sibling = node.parent.leftChild 
+            end
+
+            if !sibling.rightChild.color && !sibling.leftChild.color
+                sibling.color = true
+                node = node.parent
+            else
+                if !sibling.leftChild.color
+                    sibling.rightChild.color = false
+                    sibling.color = true
+                    left_rotate!(tree, sibling)
+                    sibling = node.parent.leftChild
+                end
+
+                sibling.color = node.parent.color
+                node.parent.color = false
+                sibling.leftChild.color = false
+                right_rotate!(tree, node.parent)
+                node = tree.root
+            end
+        end
+    end
+    node.color = false
+    return nothing
+end
+
+"""
+    rb_transplant(tree::RBTree, u::Union{RBTreeNode, Nothing}, v::Union{RBTreeNode, Nothing})
+
+Replaces `u` by `v` in the `tree` and updates the `tree` accordingly.
+"""
+function rb_transplant(tree::RBTree, u::Union{RBTreeNode, Nothing}, v::Union{RBTreeNode, Nothing})
+    if u.parent == nothing
+        tree.root = v
+    elseif u == u.parent.leftChild
+        u.parent.leftChild = v
+    else
+        u.parent.rightChild = v
+    end
+    v.parent = u.parent
+end
+
+"""
+   minimum_node(tree::RBTree, node::RBTreeNode) 
+
+Returns the RBTreeNode with minimum value in subtree of `node`. 
+"""
+function minimum_node(tree::RBTree, node::RBTreeNode)
```

Comment:
Should this overload `Base.minimum` ?
Or does "minimum value" not mean what i think it does?

Reply:
I think, it shouldn't overload `Base.minimum`. Minimum value in a subtree of a node isn't that obvious, so a separate method, should be better.